scripts/util/simulate_expression.R

```diff
@@ -4,264 +4,108 @@
-  # n - integer indicating the total number of samples
-  # mean_df - matrix of means describing groups
-  # sd_df - matrix of standard deviations describing groups
-  #         for mean and sd, ncol = number of features, nrow = number of groups
-  # r - the number of random noise parameters
-  # func_list - each element in the list stores a function to apply to a
-  #             random noise sampling (each element indicates a single param)
-  # b - the number of presence/absence features (independent from group)
-  #     (value is either 0, or is sampled from a standard normal)
-  # cell_type_mean_df - matrix of means describing artificial cell-types
-  # cell_type_sd_df - matrix of standard deviations describing cell-types
-  #       Each row represents different cell types (only 2 currently supported)
-  #       Each column represents features describing the cell types
-  # seed - add random seed as required argument
-  # zero_one_normalize - boolean to zero one normalize simulated features
-  # concat - boolean to return combined output if TRUE, defaults to list output
-  #
-  # Return:
-  # List of length 2: The first element is the simulated data matrix
-  #                   The second element is important metadata including group
-  #                       membership, cell type proportion, and the domain of
-  #                       the input functions.
+  gene_labels <- x_1$allLabels
   
-  require(tibble)
-  require(dplyr)
-
-  set.seed(seed)
-
-  # Extract Group Features
-  if (sum(mean_df + sd_df) != 0) {
-    if (all(dim(mean_df) != dim(sd_df))) {
-      stop("provide the same number of mean and standard deviation parameters")
-    } else {
-      group_params <- sampleGroupMatrix(n, mean_df, sd_df)
-      group_df <- group_params[[1]]
-      group_info <- group_params[[2]]
-    } 
-  } else {
-      group_df <- c()
-      group_info <- c()
-  }
-
-  # Get Random Noise Features
-  rand_df <- c()
-  if (r > 0) {
-    for (rand_idx in 1:r) {
-      rand_vector <- runif(n, min = 0, max = 1)
-      rand_df <- cbind(rand_df, rand_vector)
-    }
-    colnames(rand_df) <- paste('random', seq(1, ncol(rand_df)), sep = "_")
-    rand_df <- tibble::as_data_frame(rand_df)
-  }
-
-  # Get Continuous Function Features
-  cont_df <- c()
-  cont_other_df <- c()
-  if (length(func_list) > 0) {
-    for (cont_idx in 1:length(func_list)) {
-      continuous_rand_x <- runif(n, min = -1, max = 1)
-      continuous_rand_y <- func_list[[cont_idx]](continuous_rand_x)
-      
-      cont_df <- cbind(cont_df, continuous_rand_y)
-      cont_other_df <- cbind(cont_other_df, continuous_rand_x)
-    }
-    colnames(cont_df) <- paste('continuous', seq(1, ncol(cont_df)), sep = "_")
-    cont_df <- tibble::as_data_frame(cont_df)
-    
-    colnames(cont_other_df) <- paste('continuous_domain',
-                                     seq(1, ncol(cont_other_df)), sep = "_")
-    cont_other_df <- tibble::as_data_frame(cont_other_df)
-  }
+  # Combine Matrix
```

Comment:
> Ah is it the leaveOut arg?

Precisely! The though being that this will enable an `A-B+D=C` assessment, asking if the arithmetic captures a module that learns the "left out" module. I will add to docs